houston/ardu/sandbox.py

```diff
@@ -213,86 +221,189 @@ def run(self, commands: Sequence[Command]) -> 'MissionOutcome':
                     logger.debug("STATE: {}".format(self.state))
                     current_time = timer()
                     time_passed = current_time - time_start
                     wp_state[last_wp[0]] = (self.state, time_passed)
                     time_start = current_time
-                    event.clear()
+                    if recorder_filename:
+                        c_str = "Command #{}: {}\n"
+                        c_str = c_str.format(last_wp[0],
+                                             commands[last_wp[0] - 1])
+                        self.recorder.write(c_str)
+                        t = threading.Thread(
+                            target=self.recorder.write_and_flush)
+                        t.start()
+                    last_wp[0] = last_wp[1]
+                    wp_event.clear()
 
             self.connection.remove_hook('reached')
+            self.unset_recorder()
+            logger.debug("Removed hook")
+
             outcomes = []
-            state_before = initial_state
-            mission_passed = True
+            wp_state[0] = (initial_state, 0.0)
+            # consider mission passed if it doesn't hit timeout
+            mission_passed = timeout
             mission_time = 0.0
             for i in range(len(commands)):
-                cmd_index = 1 + i * 2
-                time_elapsed = wp_state[cmd_index][1]
-                # FIXME this whole time elapsed thing is wrong
-                state_after = wp_state[cmd_index + 1][0]
-                command = commands[i]
-                # determine which spec the system should observe
-                spec = command.resolve(state_before, env, config)
-                postcondition = spec.postcondition
-                passed = postcondition.is_satisfied(command,
-                                                    state_before,
-                                                    state_after,
-                                                    env,
-                                                    config)
-                mission_passed = mission_passed and passed
+                cmd_index = 1 + i
+                state_before = None
+                for j in range(cmd_index - 1, -1, -1):
+                    if j in wp_state:
+                        state_before = wp_state[j][0]
+                        break
+                if cmd_index in wp_state:
+                    state_after = wp_state[cmd_index][0]
+                    time_elapsed = wp_state[cmd_index][1] - wp_state[j][1]
+                    command = commands[i]
+                    directory = 'command{}'.format(cmd_index)
+                    coverage = self.__get_coverage(directory=directory)
+                    m = recorder_filename + '.cov'
+                    with open(m, 'a') as f:
+                        f.write('Command #{}: {}\n'.format(cmd_index, command))
+                        f.write('{}\n'.format(str(coverage)))
```

Comment:
Is this format going to be easy to parse? Could you maybe add a parser to this PR?